### PR TITLE
DT-2056 Call community api for prisoner received event

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt [hunk 1]
```diff
@@ -126,3 +139,5 @@ data class ReplaceCustodyKeyDates(
 data class Conviction(val index: String, val active: Boolean, val sentence: Sentence? = null, val custody: Custody? = null)
 
 data class Sentence(val startDate: LocalDate?)
+
+data class PrisonerReceivedDetails(val nomsNumber: String, val reason: String, val source: String, val details: String)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt:143]: Not obvious but source and details are both optional (sorry we have no swagger for this)

**Review by andymarke — CHANGES_REQUESTED** [review]
Just need to make details and source optional else we will get a JSON parse error when they are null

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt:143]: I will add tests for the optional params after this fix.

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**